kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/oxford/CameraUploadProcessor.kt

```diff
@@ -40,11 +38,16 @@ class CameraUploadProcessor(
             val projectId = checkNotNull(record.data?.projectId) { "Project ID required to upload image files." }
             val userId = checkNotNull(record.data?.userId) { "Project ID required to upload image files." }
             val relativePath = Paths.get("$projectId/$userId/$TOPIC/${record.id}/$dateDirectory/$adjustedFilename.jpg")
-            val fullPath = rootPath.resolve(relativePath).normalize()
+            val fullPath = uploaderCreate().rootDirectory().resolve(relativePath).normalize()
 
-            uploaderCreate().upload(fullPath, inputStream)
+            val url = uploaderCreate().advertisedTargetUri().resolve(fullPath.toString())
+            if(uploaderCreate() is S3FileUploader) {
+                uploaderCreate().upload(relativePath, inputStream, contents.size)
+            }
+            else {
+                uploaderCreate().upload(fullPath, inputStream, contents.size)
+            }
```

Comment:
Again, just create the uploader once. If possible, do not distinguish on implementation, otherwise, why use an interface?

Reply:
Good point. I will modify it.